### PR TITLE
Rsvp li

### PR DESCRIPTION
**Pull request by SouravSinhaRoy**
https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/open-traffic-generator/models/rsvp_li/artifacts/openapi.yaml&nocors#tag/Metrics/operation/get_metrics

### CODE REVIEW

File: result/rsvplsp.yaml [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+      type: object
+      properties:
+        rsvp_router_name:
+          description: >-
+            The name of the RSVP-TE Router.
+          type: string
+          x-field-uid: 1
+        ipv4_lsps:
+          description: >-
+            Discovered IPv4LSPs of a RSVP-TE router.
```
Comment (apratimmukherjee) [inline result/rsvplsp.yaml:32]: IPv4 Point-to-Point LSPs

File: result/rsvplsp.yaml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+          type: array
+          items:
+            $ref: '#/components/schemas/RsvpIPv4Lsp.State'
+          x-field-uid: 2
+
+    RsvpIPv4Lsp.State:
+      description: >-
+        IPv4 RSVP-TE Discovered LSPs.
```
Comment (apratimmukherjee) [inline result/rsvplsp.yaml:40]: IPv4 Point-to-Point

File: result/rsvplsp.yaml [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+          x-enum:
+            resv_tear:
+              x-field-uid: 1
+            path_tear:
+              x-field-uid: 2
+            path_timeout:
+             x-field-uid: 3
+        up_time:
+          description: >-
+            The tunnel UP time in milli seconds.
```
Comment (apratimmukherjee) [inline result/rsvplsp.yaml:130]: Need to specify expectation if tunnel is in down state.

File: result/rsvplsp.yaml [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+          type: integer
+          format: int64
+          x-field-uid: 8
+
+    RsvpLsp.Ipv4Rro:
+      description: >-
+        This contains the list of MPLS RRO objects associated with the traffic engineering tunnel.
```
Comment (apratimmukherjee) [inline result/rsvplsp.yaml:137]: Order needs to be specified i.e. which is the nearest hop and which is the farthest hop.

File: result/rsvplsp.yaml [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+      type: object
+      properties:
+        address:
+          description: >-
+            The origin IPV4 address of RSVP session.
```
Comment (apratimmukherjee) [inline result/rsvplsp.yaml:142]: Probably incorrect. Should probably be the IPv4 address of the interface or router through which the LSP has transited.

File: result/rsvplsp.yaml (hunk 4 above)
Comment (apratimmukherjee) [reply result/rsvplsp.yaml:137]: Also needs to specify conditions in which it is included. e.g. if included by the ingress , this is filled in by intermediate routers in the Path message till it reaches the egress. Similarly , if egress includes this object, then the intermediate hops are included in the Resv message sent hop by hop till it reaches the ingress.

File: result/rsvplsp.yaml [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+          type: string
+          format: ipv4
+          x-field-uid: 1
+        reported_label:
+          description: >-
+            Label reported for RRO hop.
```
Comment (apratimmukherjee) [inline result/rsvplsp.yaml:148]: Included in the RRO object included in Resv if the Label Recording flag was enabled in the Session Attribute object sent in the Path Message sent from the ingress ( please confirm )

File: result/rsvplsp.yaml [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+          type: integer
+          x-field-uid: 2
+    
+    RsvpLsp.Ipv4Ero:
+      description: >-
+        This contains the list of MPLS ERO objects associated with the traffic engineering tunnel.
```
Comment (apratimmukherjee) [inline result/rsvplsp.yaml:154]: sub-objects included in the Explicit Route Object (ERO ) sent in the Path Message from the ingress.

**Review by apratimmukherjee — COMMENTED** [review]
Please include Prasenjit in the review and confirm that the proposed attributes cover all attributes to be published by gnmi

File: result/rsvplsp.yaml (hunk 1 above)
Comment (SouravSinhaRoy) [reply result/rsvplsp.yaml]: updated

Comment (SouravSinhaRoy) [reply result/rsvplsp.yaml]: updated

**Review by PrasenjitAdhikary — APPROVED** [review]

**Review by apratimmukherjee — APPROVED** [review]
Looks good to me.

**Pull request merged**